src/server/bg_services/buckets_reclaimer.js

```diff
@@ -83,6 +95,29 @@ class BucketsReclaimer {
         return system_store.data.buckets.filter(bucket => Boolean(bucket.deleting));
     }
 
+    _add_bucket_to_delete_lists(bucket) {
+        this.buckets_to_delete.push(bucket._id);
+        const tiering_policy = bucket.tiering;
+        this.tiering_to_delete.push(tiering_policy._id);
+        this.tiers_to_delete.push(...tiering_policy.tiers.map(t => t.tier._id));
+        system_store.data.accounts.forEach(account => {
+            if (!account.allowed_buckets || (account.allowed_buckets && account.allowed_buckets.full_permission)) return;
+            this.account_updates.push({
```

Comment:
@jackyalbo 
Unnessasery and heavy update query on accounts collection.
In worst-case scenario, it will trigger an update to all accounts that do not have `full_permission` even if they do not have the bucket in their `permissions_list`.
There is no reason to do this heavy lifting in Mongo, you already have all the data in the cache.
Just code another _.filter that will filter only the accounts that have the implicit permissions to the deleted buckets and then trigger the updates on these accounts.

Reply:
removed the account without the bucket in their permission list from acconut_updates